sprint_9-solution-http-api

7Jenko opened the pull request:
Настроить доступ к методам программы через HTTP API.

7Jenko committed:
sprint_9-solution-http-api
Настроить доступ к методам программы через HTTP API.

slavrikk commented on src/com/yandex/app/http/EpicHandler.java:
```diff
@@ -0,0 +1,101 @@
+    private void handlePostEpic(HttpExchange exchange) throws IOException {
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(exchange.getRequestBody(),
+                StandardCharsets.UTF_8))) {
+            String json = reader.lines().collect(java.util.stream.Collectors.joining());
```
Пакеты лучше прописывать в импортах (`java.util.stream`)

slavrikk commented on src/com/yandex/app/http/EpicHandler.java:
```diff
@@ -0,0 +1,101 @@
+import java.time.LocalDateTime;
+import java.util.List;
+
+public class EpicHandler extends BaseHttpHandler implements HttpHandler {
```
BaseHttpHandler уже имплементирует `HttpHandler`, в хендлерах интерфейс прописывать уже не нужно

slavrikk replied:
Замечание для всех хендлеров

slavrikk commented on src/com/yandex/app/http/HistoryHandler.java:
```diff
@@ -0,0 +1,48 @@
+
+public class HistoryHandler extends BaseHttpHandler implements HttpHandler {
+    private final TaskManager taskManager;
+    private final Gson gson = new GsonBuilder()
```
Где то gson создается в конструкторе, где то в переменной, лучше привести к единому стилю

slavrikk commented on test/HttpTaskServerTest.java:
```diff
@@ -0,0 +1,87 @@
+    @Test
+    void testGetAllTasks() throws IOException {
+        int responseCode = sendRequest("GET", "/tasks");
+        assertEquals(HttpURLConnection.HTTP_OK, responseCode);
```
Для методов get хорошо бы добавить тестовые данные, и проверить, что таски возвращаются корректно

7Jenko committed:
sprint_9-solution-http-api
Перенести пакеты в импорты  (java.util.stream);
Убрать интерфейс HttpHandler из всех хэндлеров;
Привести gson к единому стилю;
Добавить тестовые данные для методов GET.
